lms/djangoapps/ccx/tests/test_overrides.py

```diff
@@ -129,3 +139,143 @@ def test_override_is_inherited_even_if_set_in_mooc(self):
         override_field_for_ccx(self.ccx, chapter, 'due', ccx_due)
         vertical = chapter.get_children()[0].get_children()[0]
         self.assertEqual(vertical.due, ccx_due)
+
+
+class FuzzyInt(int):
+    """
+    A subclass of integer that checks for equality over a range. This is useful
+    for assertions that don't take a range, such as Django's assertNumQueries.
+
+    http://lukeplant.me.uk/blog/posts/fuzzy-testing-with-assertnumqueries/
+    """
+    def __new__(cls, high, low):
+        obj = super(FuzzyInt, cls).__new__(cls, high)
+        obj.low = low
+        obj.high = high
+        return obj
+
+    def __eq__(self, other):
+        return self.low <= other <= self.high
+
+    def __repr__(self):
+        return '[{}..{}]'.format(self.low, self.high)
+
+
+@attr('shard_1')
+@mock.patch.dict('django.conf.settings.FEATURES', {'ENABLE_XBLOCK_VIEW_ENDPOINT': True})
+@ddt.ddt
+class TestFieldOverridePerformance(ModuleStoreTestCase):
+    """
+    Tests that ensure the field
+    """
+    def setUp(self):
+        """
+        Create a test client, course, and user.
+        """
+        super(TestFieldOverridePerformance, self).setUp()
+
+        self.request_factory = RequestFactory()
+        self.student = UserFactory.create()
+        self.request = self.request_factory.get("foo")
+        self.request.user = self.student
+
+        MakoMiddleware().process_request(self.request)
+
+    def setup_course(self, course_name):
+        """
+        Imports some XML course data.
+        """
+        course = import_course_from_xml(
+            self.store,
+            999,
+            DATA_DIR,
+            ['test_increasing_size/graded_{}'.format(course_name)]
+        )[0]
+
+        CourseEnrollment.enroll(self.student, course.id)
+
+        return course
+
+    def grade_course(self, course):
+        """
+        Renders the progress page for the given course.
+        """
+        return progress(
+            self.request,
+            course_id=course.id.to_deprecated_string(),
+            student_id=self.student.id
+        )
+
+    def instrument_course_grading(self, course_name, queries, reads):
+        """
+        Renders the progress page, instrumenting Mongo reads and SQL queries.
+        """
+        course = self.setup_course(course_name)
+        with self.assertNumQueries(FuzzyInt(queries - 1, queries + 1)):
```

Comment:
Alright, I've removed it in favor of caching. Tests are passing locally but hopefully things will be alright on Jenkins